app/models/ext_management_system.rb

```diff
@@ -426,6 +426,24 @@ def enable!
     update!(:enabled => true)
   end
 
+  # Wait until all associated workers are dead to destroy this ems
+  def orchestrate_destroy
+    disable! if enabled?
+
+    workers = MiqWorker.find_current_or_starting.where(:queue_name => "ems_#{id}") #better way to get 'queue_name'?
```

Comment:
As a followup PR, can you do this?

https://github.com/ManageIQ/manageiq/blob/d199f44cd9a1ee5202f14222a9c74ed3f5fc9c7f/app/models/mixins/per_ems_worker_mixin.rb#L94-L101

Change:
```ruby
def queue_name_for_ems(ems)
  # Host objects do not have dedicated refresh workers so request a generic worker which will
  # be used to make a web-service call to a SmartProxy to initiate inventory collection.
  return "generic" if ems.kind_of?(Host) && ems.acts_as_ems?
  
  return ems unless ems.kind_of?(ExtManagementSystem)
  "ems_#{ems.id}"
end
```

To:
```ruby
def queue_name_for_ems(ems)
  # Host objects do not have dedicated refresh workers so request a generic worker which will
  # be used to make a web-service call to a SmartProxy to initiate inventory collection.
  return "generic" if ems.kind_of?(Host) && ems.acts_as_ems?

  return ems unless ems.kind_of?(ExtManagementSystem)
  ems.queue_name
end
```

And in ExtManagementSystem:
```ruby
def queue_name
  "ems_#{ems.id}"
end
```

Finally... then this line becomes:

```ruby
workers = MiqWorker.find_current_or_starting.where(:queue_name => ems.queue_name) 
```